### PR TITLE
Serverless deployment configuration

### PR DESCRIPTION
**Pull request by thinkharderdev**
Add serverless configs for provisioning Aurora RDS cluster as part of our serverless stack.

DB migrations are packaged as a separate lambda function which is invoked after deployments (this piece is not working yet, trying to debug it)

### CODE REVIEW

File: backend/alembic/env.py [hunk 1]
```diff
@@ -34,6 +42,16 @@ def get_database_url_config_key():
...
+def get_db_url():
+    application_environment = os.getenv('APP_ENV')
+    if application_environment == 'tilt':
+        return 'postgresql+pygresql://app:apppassword@postgres:5432/housinghub'
+    elif application_environment == 'dev':
+        return 'postgresql+auroradataapi://:@/housinghubdev'
+    elif application_environment == 'prod':
+        return 'postgresql+auroradataapi://:@/housinghubprod'
+    else:
+        return 'postgresql+pygresql://app:apppassword@0.0.0.0:5432/housinghub'
```
Comment (dgoldstein1) [inline backend/alembic/env.py:54]:
```suggestion
  return os.getenv("DB_URL)
```

File: backend/alembic/migration.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from alembic.config import Config
+from alembic import command
+import inspect
+import os
+
+
+def main(event, context):
+    _current_dir = os.path.dirname(os.path.abspath(inspect.stack()[0][1]))
```
Comment (dgoldstein1) [inline backend/alembic/migration.py:8]:
```suggestion
    _current_dir = os.getcwd()
```
is _current_dir ever used?

Comment (thinkharderdev) [reply backend/alembic/migration.py:8]: Yeah, it's used on line 11:

```
    alembic_cfg.set_main_option('script_location', _current_dir)
```

We actually don't want os.getcwd(). That gives the current dir of the user's shell. We need the location of the handler.py file.

**Comment by thinkharderdev** [comment]
> looking good so far-- is your goal to deploy this whole stack with serverless? If so can you add the readme the instructions needed to deploy?

Yep. READMEs are updated with deploy instructions.

File: backend/README.md [hunk 3]
```diff
@@ -151,3 +151,50 @@ $(cd alembic && alembic upgrade head)
 serverless wsgi serve
 ```
 
+### Deployment 
+
+This API is deployed as a serverless application on AWS. The entire stack deployment is handled but the serverless configuration. **However**, the database migrations are not done as part of the deployment and have to be run separately from the serverless deployment (see details below).
```
Comment (dgoldstein1) [inline backend/README.md:156]: The entire stack is handled `with` (?) the serverless configuration

Comment (thinkharderdev) [reply backend/README.md]: Haha, nice. Thanks for the proofread!

Yeah, I deployed the current dev environment using these configs. I still need to setup a CloudFront distribution to front the S3 bucket but will do that as a one-time out-of-band setup.

**Pull request merged**